contracts/contracts/strategies/BaseCurveStrategy.sol

```diff
@@ -108,53 +117,71 @@ abstract contract BaseCurveStrategy is InitializableAbstractStrategy {
     ) external override onlyVault nonReentrant {
         require(_amount > 0, "Invalid amount");
 
-        emit Withdrawal(_asset, address(assetToPToken[_asset]), _amount);
+        emit Withdrawal(_asset, pTokenAddress, _amount);
 
-        (uint256 contractPTokens, , uint256 totalPTokens) = _getTotalPTokens();
+        uint256 contractCrv3Tokens = IERC20(pTokenAddress).balanceOf(
+            address(this)
+        );
 
         uint256 coinIndex = _getCoinIndex(_asset);
         int128 curveCoinIndex = int128(uint128(coinIndex));
         // Calculate the max amount of the asset we'd get if we withdrew all the
         // platform tokens
         ICurvePool curvePool = ICurvePool(platformAddress);
+
+        uint256 virtual_price = curvePool.get_virtual_price();
+        /**
+         * Add a 5% threshold to help calculate required amount of crv3Tokens:
+         *  - convert asset to 18 decimals
+         *  - divide by virtual price to get 3Crv equivalent
+         *  - add a 5% threshold to it
+         */
+        // slither-disable-next-line divide-before-multiply
+        uint256 crv3TokensTreshold = (((_amount.scaleBy(
+            18,
+            Helpers.getDecimals(_asset)
+        ) * 105) * 1e18) / virtual_price) / 100;
```

Comment:
Will something bad happen here because of this 105% and a peg loss on 3crv?

Reply:
This 105 is basically a somewhat non-intuitive hardcoded slippage limit. We should probably use our normal 3pool slippage limit here.

Reply:
Addressing this issue in this PR: https://github.com/OriginProtocol/origin-dollar/pull/1161